bots/pr/src/main/java/org/openjdk/skara/bots/pr/BranchCommand.java

```diff
@@ -0,0 +1,133 @@
+
+    private void showHelp(PrintWriter reply) {
+        reply.println("Usage: `/branch <name>`");
+    }
+
+    @Override
+    public String description() {
+        return "create a branch";
+    }
+
+    @Override
+    public String name() {
+        return branch.name();
+    }
+
+    @Override
+    public boolean allowedInCommit() {
+        return true;
+    }
+
+    @Override
+    public boolean allowedInPullRequest() {
+        return false;
+    }
+
+    @Override
+    public void handle(PullRequestBot bot, HostedCommit commit, LimitedCensusInstance censusInstance,
+            ScratchArea scratchArea, CommandInvocation command, List<Comment> allComments, PrintWriter reply) {
+        try {
+            if (!bot.integrators().contains(command.user().username())) {
+                reply.println("Only integrators for this repository are allowed to use the `/branch` command.");
+                return;
+            }
+            if (censusInstance.contributor(command.user()).isEmpty()) {
+                reply.println("To use the `/branch` command, you need to be in the OpenJDK [census](https://openjdk.org/census)"
+                        + " and your GitHub account needs to be linked with your OpenJDK username"
+                        + " ([how to associate your GitHub account with your OpenJDK username]"
+                        + "(https://wiki.openjdk.org/display/skara#Skara-AssociatingyourGitHubaccountandyourOpenJDKusername)).");
+                return;
+            }
+
+            var args = command.args();
+            if (args.isBlank()) {
+                showHelp(reply);
+                return;
+            }
+
+            var parts = args.split(" ");
+            if (parts.length > 1) {
+                showHelp(reply);
+                return;
+            }
+            var branchName = parts[0];
+
+            var localRepoDir = scratchArea.get(this)
+                    .resolve(bot.repo().name());
+            var localRepo = bot.hostedRepositoryPool()
+                               .orElseThrow(() -> new IllegalStateException("Missing repository pool for PR bot"))
+                               .materialize(bot.repo(), localRepoDir);
+            localRepo.fetch(bot.repo().authenticatedUrl(), commit.hash().toString(), true);
+
+            var remoteBranches = localRepo.remoteBranches(bot.repo().authenticatedUrl().toString());
+            var remoteBranchNames = remoteBranches.stream()
+                                                  .map(Reference::name)
+                                                  .collect(Collectors.toSet());
+            if (remoteBranchNames.contains(branchName)) {
+                var ref = remoteBranches.stream().filter(r -> r.name().equals(branchName)).findFirst();
+                if (ref.isEmpty()) {
+                    throw new IllegalStateException("Cannot find hash for remote branch '" + branchName + "'");
+                }
```

Comment:
I think this turned out a little awkward, especially the exception that should never actually happen.
```suggestion
            var existingBranch = remoteBranches.stream()
                    .filter(b -> b.name().equals(branchName))
                    .findAny();
            if (existingBranch.isPresent()) {
```

Reply:
Sure, I rewrote it a bit in the latest version.